File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        StringBuilder answer = new StringBuilder();
        boolean mode = false;

        for (int i=0; i<code.length(); i++) {
            if (mode) {
                if (code.charAt(i) == '1') {
                    mode = false;
                    continue;
                }
                if (i%2 == 1) answer.append(code.charAt(i));
            } else {
                if (code.charAt(i) == '1') {
                    mode = true;
                    continue;
                }
                if (i%2 == 0) answer.append(code.charAt(i));
            }
        }

        if (answer.toString().isEmpty()) return "EMPTY";
        return answer.toString();
    }
}

File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    mode = 0
    
    for i in range(0, len(code)):
        if code[i]=='1':
            mode = 1 - mode
        elif mode==0 and i%2==0:
            answer += code[i]
        elif mode==1 and i%2==1:
            answer += code[i]
    
    if answer=="": 
        return "EMPTY" 
    
    return answer